src/mlpack/methods/cf/decomposition_policies/quic_svd_method.hpp

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file methods/cf/decomposition_policies/quic_svd_method.hpp
+ * @author Adarsh Santoria
+ *
+ * Implementation of the quic svd method for use in
+ * Collaborative Fitlering.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_CF_DECOMPOSITION_POLICIES_QUIC_SVD_METHOD_HPP
+#define MLPACK_METHODS_CF_DECOMPOSITION_POLICIES_QUIC_SVD_METHOD_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/quic_svd/quic_svd.hpp>
+
+namespace mlpack {
+
+/**
+ * Implementation of the Quic SVD policy to act as a wrapper when
+ * accessing Quic SVD from within CFType.
+ *
+ * An example of how to use QuicSVDPolicy in CF is shown below:
+ *
+ * @code
+ * extern arma::mat data; // data is a (user, item, rating) table.
+ * // Users for whom recommendations are generated.
+ * extern arma::Col<size_t> users;
+ * arma::Mat<size_t> recommendations; // Resulting recommendations.
+ *
+ * CFType<QuicSVDPolicy> cf(data);
+ *
+ * // Generate 10 recommendations for all users.
+ * cf.GetRecommendations(10, recommendations);
+ * @endcode
+ */
+class QuicSVDPolicy
+{
+ public:
+  /**
+   * Use quic SVD method to perform collaborative filtering
+   */
+  QuicSVDPolicy()
+  {
+    /* Nothing to do here */
+  }
+
+  /**
+   * Apply Collaborative Filtering to the provided data set using the
+   * quic SVD.
+   *
+   * @param data Data matrix: dense matrix (coordinate lists)
+   *    or sparse matrix(cleaned).
+   * @param * (cleanedData) item user table in form of sparse matrix.
+   * @param * (rank) Rank parameter for matrix factorization.
+   * @param * (maxIterations) Maximum number of iterations.
+   * @param * (minResidue) Residue required to terminate.
+   * @param * (mit) Whether to terminate only when maxIterations is reached.
+   */
+  template<typename MatType>
+  void Apply(const MatType& data,
+             const arma::sp_mat& /* cleanedData */,
+             const size_t /* rank */,
+             const size_t /* maxIterations */,
+             const double /* minResidue */,
+             const bool /* mit */)
+  {
+    arma::mat sigma;
+
+    // Do singular value decomposition using the quic SVD algorithm.
+    QUIC_SVD quicsvd;
+    quicsvd.Apply(data, w, h, sigma);
+
+   if (data.n_cols < data.n_rows)
+  {
+    arma::mat tempMat = w;
+    w = h;
+    h = tempMat;
+  }
```

Comment:
Can you fix the style here please?  I'm guessing that there are some tab characters.  mlpack code should be all spaces (no tabs).  See also https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines :+1: